### PR TITLE
refactor(Space): less to cssinjs & add compact mode

### PR DESCRIPTION
**Pull request by buqiyuan**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/button/button.tsx [hunk 1]
```diff
@@ -49,6 +50,10 @@ export default defineComponent({
     const hasTwoCNChar = ref(false);
 
     const autoInsertSpace = computed(() => autoInsertSpaceInButton.value !== false);
+    const { compactSize, compactItemClassnames } = useCompactItemContext(
+      prefixCls.value,
```
Comment (tangjinzhou) [inline components/button/button.tsx:54]: 丢失响应式了

File: components/space/Compact.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import { computed, defineComponent } from 'vue';
+import type { PropType, ExtractPropTypes } from 'vue';
+import PropTypes from '../_util/vue-types';
+import { tuple } from '../_util/type';
+import { isEmpty } from 'lodash-es';
+
+export const spaceCompactItemProps = () => ({
+  compactSize: String as PropType<SizeType>,
+  compactDirection: PropTypes.oneOf(tuple('horizontal', 'vertical')).def('horizontal'),
+  isFirstItem: { type: Boolean, default: undefined },
```
Comment (tangjinzhou) [inline components/space/Compact.tsx:16]: 有个 booleanType

File: components/space/Compact.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    const clx = computed(() => {
+      return classNames(prefixCls.value, hashId.value, {
+        [`${prefixCls.value}-rtl`]: directionConfig.value === 'rtl',
+        [`${prefixCls.value}-block`]: props.block,
+        [`${prefixCls.value}-vertical`]: props.direction === 'vertical',
+      });
+    });
+
+    const nodes = computed(() => {
+      const compactItemContext = SpaceCompactItemContext.useInject();
```
Comment (tangjinzhou) [inline components/space/Compact.tsx:99]: useInject 不能放 computed 里吧

File: components/space/Compact.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      const childNodes = slots.default?.() || [];
```
Comment (tangjinzhou) [inline components/space/Compact.tsx:100]: slots 无法响应式更新，nodes直接放 return 里面好了

File: components/space/index.tsx [hunk 5]
```diff
@@ -110,7 +114,7 @@ const Space = defineComponent({
               }
             }
 
-            return (
+            return wrapSSR(
```
Comment (tangjinzhou) [inline components/space/index.tsx:117]: inheritAttrs: false, 手动挂载 attrs

**Review by tangjinzhou — CHANGES_REQUESTED** [review]

File: components/space/index.zh-CN.md [hunk 6]
```diff
@@ -15,11 +15,37 @@ cover: https://gw.alipayobjects.com/zos/antfincdn/wc6%263gJ0Y8/Space.svg
...
-| direction | 间距方向 | `vertical` \| `horizontal`                 | `horizontal` | 1.6.5 |
-| size      | 间距大小 | `small` \| `middle` \| `large` \| `number` | `small`      | 1.6.5 |
+### Space
+
+| 参数 | 说明 | 类型 | 默认值 | 版本 |
+| --- | --- | --- | --- | --- |
+| align | 对齐方式 | `start` \| `end` \|`center` \|`baseline` | - | 1.6.5 |
+| direction | 间距方向 | `vertical` \| `horizontal` | `horizontal` | 1.6.5 |
+| size | 间距大小 | `small` \| `middle` \| `large` \| `number` | `small` | 1.6.5 |
+| split | 设置拆分 | VueNode \| v-slot | - | 1.6.5 |
```
[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/space/index.zh-CN.md:29]: 这个好像是之前就有的特性，只是文档中没有写，不是新增的特性

[the remark this answers is not in the document]
Comment (buqiyuan) [reply components/space/index.zh-CN.md:29]: 可能是这个 [PR](https://github.com/vueComponent/ant-design-vue/pull/4171/files#diff-d42c3072a5016964aad21722cb773b77e6fc5824adf56dc16efaecff82aab948) 的时候加的

**Pull request merged**